Chap_API_Event.tex

```diff
@@ -415,15 +423,15 @@ \subsection{\code{PMIx_Deregister_event_handler}}
 
 If \refarg{cbfunc} is \code{NULL}, the function will be treated as a \emph{blocking} call and the result of the operation returned in the status code.
 
-If \refarg{cbfunc} is non-\code{NULL}, the function will be treated as a \emph{non-blocking} call and return one of the following:
+If \refarg{cbfunc} is non-\code{NULL}, the function will be treated as a \emph{non-blocking} call.
+
+\returnsimplenb
 
 \begin{itemize}
-\item \refconst{PMIX_SUCCESS}, indicating that the request is being processed - result will be returned in the provided \refarg{cbfunc}. Note that the library must not invoke the callback function prior to returning from the \ac{API}.
-\item \refconst{PMIX_OPERATION_SUCCEEDED}, indicating that the request was immediately processed and returned \textit{success} - the \refarg{cbfunc} will \textit{not} be called
-\item a PMIx error constant indicating either an error in the input or that the request was immediately processed and failed - the \refarg{cbfunc} will \textit{not} be called
+\item \refconst{PMIX_OPERATION_SUCCEEDED}, returned only when \refarg{cbfunc} is non-\code{NULL} and the request was immediately processed successfully - the \refarg{cbfunc} will \textit{not} be called
```

Comment:
I'm afraid this is incorrect - the `cbfunc` is irrelevant here. If the operation is atomically completed, you get `OPERATION_SUCCEEDED` regardless of NULL or non-NULL cbfunc.

Reply:
Correct - of course, it can always immediately return an error code. If it returns SUCCESS, then you can still get an error code in the cbfunc - all "success" means is that there was no immediate reason to reject the request.

Reply:
Added a revision exception for this change.